src/mcmc/algorithm.jl

```diff
@@ -11,4 +12,35 @@ this wrapping occurs automatically.
 """
 abstract type InferenceAlgorithm end
 
+# TODO(penelopeysm): remove
 DynamicPPL.default_chain_type(sampler::Sampler{<:InferenceAlgorithm}) = MCMCChains.Chains
+
+"""
+    update_sample_kwargs(spl::AbstractSampler, N::Integer, kwargs)
+
+Some samplers carry additional information about the keyword arguments that
+should be passed to `AbstractMCMC.sample`. This function provides a hook for
+them to update the default keyword arguments. The default implementation is for
+no changes to be made to `kwargs`.
+"""
```

Comment:
I think it's worth adding that this function returns a `NamedTuple` (if I understand correctly).

Reply:
also, it's prob worth mentioning what is `N`?